test/hotspot/gtest/opto/test_constant_division.cpp

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+
+#include <random>
+#include <type_traits>
+#include <vector>
```

Comment:
We mostly don't use C++ standard library facilities in HotSpot, even in tests; see the style guide.
<type_traits> is permitted.  We have GrowableArray instead of <vector>.  And we have os::random,
unless this really needs something better (seems unlikely, other than needing to deal with wider
types than int.)  Also, stdlib includes at the end again (except I think unittest.hpp is supposed to
_really_ be last.)

Reply:
I see, for `vector` I did not look it up carefully and thought that we need a VM for `GrowableArray`s, and `<random>` is used mainly for wider types and custom bounds. I have changed them to use what are offered by OpenJDK instead.